pymongoexplain/__main__.py

```diff
@@ -13,48 +13,35 @@
 # limitations under the License.
 
 
-from pymongo import monitoring, MongoClient
 from pymongo.collection import Collection
+from .explainable_collection import ExplainCollection
+
 import sys
-from bson.son import SON
+import logging
 
-class CommandLogger(monitoring.CommandListener):
-    def __init__(self):
-        self.payloads = []
+FORMAT = '%(asctime)s %(levelname)s %(module)s %(message)s'
+logging.basicConfig(format=FORMAT, level=logging.INFO)
 
-    def started(self, event):
-        print("started")
-        self.payloads.append(event.command)
+old_function_names = ["update_one", "replace_one", "update_many", "delete_one",
+                      "delete_many", "aggregate", "watch", "find", "find_one",
+                      "find_one_and_delete", "find_one_and_replace",
+                      "find_one_and_update", "count_documents",
+                      "estimated_document_count", "distinct"]
+old_functions = [getattr(Collection, i) for i in old_function_names]
 
-    def succeeded(self, event):
-        pass
 
-    def failed(self, event):
-        pass
+def make_func(old_func, old_func_name):
+    def new_func(self: Collection, *args, **kwargs):
+        logging.info("%s explain response: %s"% (old_func_name, getattr(
+            ExplainCollection(self),old_func_name)(*args, **kwargs)))
```

Comment:
I suggest refactoring this slightly to make the code a bit more readable:
```
res = getattr(ExplainCollection(self), old_func_name)(*args, **kwargs)
logging.info("%s explain response: %s", old_func_name, res) 
```
Note that with the `logging` methods you should pass the format string arguments without using `%` on the format string directly. See https://docs.python.org/3/howto/logging.html#logging-variable-data